library/os_network.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/python
+
+# Copyright (c) 2014 Hewlett-Packard Development Company, L.P.
+# Copyright (c) 2013, Benno Joy <benno@ansible.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+from distutils.version import StrictVersion
+
+
+DOCUMENTATION = '''
+---
+module: os_network
+short_description: Creates/removes networks from OpenStack
+extends_documentation_fragment: openstack
+version_added: "2.0"
+author: "Monty Taylor (@emonty)"
+description:
+   - Add or remove network from OpenStack.
+options:
+   name:
+     description:
+        - Name to be assigned to the network.
+     required: true
+   shared:
+     description:
+        - Whether this network is shared or not.
+     required: false
+     default: false
+   admin_state_up:
+     description:
+        - Whether the state should be marked as up or down.
+     required: false
+     default: true
+   external:
+     description:
+        - Whether this network is externally accessible.
+     required: false
+     default: false
+   state:
+     description:
+        - Indicate desired state of the resource.
+     choices: ['present', 'absent']
+     required: false
+     default: present
+   provider_physical_network:
+     description:
+        - The physical network where this network object is implemented.
+     required: false
+     default: None
+     version_added: "2.1"
+   provider_network_type:
+     description:
+        - The type of physical network that maps to this network resource.
+     choices: ['flat', 'vlan', 'vxlan', 'gre', 'uplink']
```

Comment:
This version you've brought in doesn't have the change in it to allow for a `local` network type. You might need to get it from the devel/ branch instead of the 2.2 branch.